UNIXFS.md

```diff
@@ -192,6 +200,21 @@ This scheme would see metadata stored in an external database.
 
 The downsides to this are that metadata would not be transferred from one node to another when syncing as [Bitswap] is not aware of the database, and in-tree metadata
 
+### TimeSpec protobuf datatype rationale
+
+#### EpochSeconds
+
+The integer portion of the epoch is represented on the wire using a varint encoding. While this is inefficient for
+negative values, it avoids introducing zig-zag encoding. Negative epoch values will be exceedingly rare, and there
+could very well be value in having such cases stand out, while at the same keeping the "usual" positive values easy
+to eyeball. The varint representing the time of writing this text is 5 bytes long. It will remain so until
+October 26, 3058 ( 34,359,738,367 )
+
+#### EpichNanoseconds
+Since fractional values will very often be > 2^28 nanoseconds, that part is represented as a 4-byte `fixed32`,
```

Comment:
Not sure I understand the maths behind why they’ll often be this big?

Reply:
The nanosecond part is essentially a random number between 0 and 1e9.

* 1/2 the time, you'll need 29 bits (`ceil(logtwo(1e9/2))`) to represent this. That's 4 bytes when using fixed32 and 5 bytes when using a varint.
* 99% of the time, you'll need at least 23 bits to represent it (`logtwo(1e9/100)`). That's 4 bytes when using a varint.

Reply:
@alanshaw sorry for the delay answering, but yes: what they said ;)